core/modules/filter/filter.module

```diff
@@ -1342,12 +1342,19 @@ function filter_process_format($element) {
       '#parents' => array_merge($element['#parents'], array('format')),
     );
   }
+  // If there's only a single text format available to the current user, then
+  // render it as a hidden form element.
   else {
     $element['format']['format'] = array(
       '#type' => 'hidden',
       '#value' => $element['#format'],
       '#default_value' => $element['#format'],
-      '#attributes' => array('class' => array('filter-list')),
+      '#attributes' => array(
+        'class' => array('filter-list'),
+        // Add the human-readable name of the text format, so it can be used as
+        // the fieldset summary.
+        'data-text-format-name' => array($options[$element['#format']]),
```

Comment:
I don't think this needs to be an array here. When output, this creates `data-text-format-name=["Formatted HTML"]`. Additionally, this needs to be filtered through check_plain().

```suggestion
        'data-text-format-name' => check_plain($options[$element['#format']]),
```

Reply:
Done: [`edb5555` (#2550)](https://github.com/backdrop/backdrop/pull/2550/commits/edb5555dcd328c2318c5ad3e0ba1fbc1801cba7d)